crates/cairo-lint-core/src/fix.rs

```diff
@@ -231,4 +231,42 @@ impl Fixer {
 
         node.get_text(db).to_string()
     }
+
+    /// Attempts to fix a collapsible if-statement by combining its conditions.
+    /// This function detects nested `if` statements where the inner `if` can be collapsed
+    /// into the outer one by combining their conditions with `&&`. It reconstructs the
+    /// combined condition and the inner block, preserving the indentation and formatting.
+    ///
+    /// # Arguments
+    ///
+    /// * `db` - A reference to the `SyntaxGroup`, which provides access to the syntax tree.
+    /// * `node` - A `SyntaxNode` representing the outer `if` statement that might be collapsible.
+    ///
+    /// # Returns
+    ///
+    /// A `String` containing the fixed code with the combined conditions if a collapsible
+    /// `if` is found. If no collapsible `if` is detected, the original text of the node is
+    /// returned.
+
+    pub fn fix_collapsible_if(&self, db: &dyn SyntaxGroup, node: SyntaxNode) -> String {
```

Comment:
you need to call this function in the big match in the `fix_plugin_diagnostics` if i recall correctly